From 19d7db0e354566e0dd0ea73b1e38fc715c19233b Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 9 Jun 2024 23:28:03 +0200
Subject: [PATCH 17/18] Workaround `Unknown CMake command
 "_qt_test_emscripten_version".`

Not sure why `QtPublicWasmToolchainHelpers.cmake` is not included
automatically in my WASM builds; this helps fixing the configuration error
at least.

Change-Id: Id63c90a1f564688849c5bd7193195faa1eb38a0a
---
 src/corelib/Qt6WasmMacros.cmake | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/corelib/Qt6WasmMacros.cmake b/src/corelib/Qt6WasmMacros.cmake
index 1de6e5448c7..a7a73195f21 100644
--- a/src/corelib/Qt6WasmMacros.cmake
+++ b/src/corelib/Qt6WasmMacros.cmake
@@ -1,6 +1,11 @@
 # Copyright (C) 2022 The Qt Company Ltd.
 # SPDX-License-Identifier: BSD-3-Clause
 
+# Workaround `Unknown CMake command "_qt_test_emscripten_version".`
+if(NOT COMMAND _qt_test_emscripten_version)
+    include("$ENV{EMSDK}/system/lib/cmake/Qt6/QtPublicWasmToolchainHelpers.cmake")
+endif()
+
 # Copy in Qt HTML/JS launch files for apps.
 function(_qt_internal_wasm_add_target_helpers target)
 
-- 
2.45.2

